lowcar/devices/KoalaBear/KoalaBear.cpp

```diff
@@ -241,21 +287,17 @@ void KoalaBear::device_enable() {
 }
 
 void KoalaBear::device_disable() {
-    // digitalWrite(SLEEP, HIGH);
-    //this->pid->setCoefficients(1, 0, 0);
-    //this->pid->resetEncoder();
-
-    this->pid_a->set_coefficients(0.0, 0.0, 0.0);
-    this->pid_b->set_coefficients(0.0, 0.0, 0.0);
+    this->pid_a->set_coefficients(KP_DEFAULT, KI_DEFAULT, KD_DEFAULT);
+    this->pid_b->set_coefficients(KP_DEFAULT, KI_DEFAULT, KD_DEFAULT);
 
     this->target_speed_a = 0.0;
     this->target_speed_b = 0.0;
-    this->pid_enabled_a = FALSE;
-    this->pid_enabled_b = FALSE;
+    this->manual_enabled_a = FALSE;
+    this->manual_enabled_b = FALSE;
 }
 
 void KoalaBear::device_actions() {
-    float target_A, target_B;
+    float command_a, command_b, actual_target_a, actual_target_b;
```

Comment:
So if I'm understanding correctly,
1. Student writes to `duty_cycle` with a value in [-1.0, 1.0]
2. `Device::device_write()` writes that exact value to `this->target_speed`
3. `actual_target_speed` is `this->target_speed` adjusted for `MAX_DUTY_CYCLE`, acceleration, and invertedness
4. `command` is `actual_target_speed` adjusted with PID control (if applicable)
5. `command` is the _actual_ duty cycle written to the pin

I was thinking it might(?) be clearer if the names were:
- param `duty_cycle` --> `target_speed`
- `this->target_speed` -- keep the same (student writing to `target_speed` writes to `target_speed`)
- `this->actual_target_speed` -- keep the same or `this->adjusted_target_speed`
- command --> `duty_cycle`

Are we still considering changing the "speed" units to encoder ticks per second in the future?

Reply:
Yes, what you've got there is exactly right and I'll change it now. The "speed" units I think we're gonna keep it to -1.0 to 1.0 still, since I think it's a bit more intuitive for students than the somewhat arbitrary range -2700.0 to 2700.0 "ticks per second" (what's a tick? why can't it go higher than 2700 tps? etc.).